Basic push notifications android

zemiret opened the pull request:
Basic push notifications setup for android

osdnk commented:
I strongly appreciate your job! It is amazing. Except some _aesthetic_ code issue (use Eslint for God's sake! :) ). I'm extremely thankful for this research. Well, I am missing chance to make notifications handler by myself again :).
I would be ever more satisfied, If you would like to read about:
1. Handling _offline_ nots. Just to remind about lecture :). It should be easy one. 
2. Read about customization. We wish to set SFI logo into not.

However, I'm not going to merge it now because:
1. It's not crucial for further development. 
2. Let us wait for iOS tests. I have iMac for the weekend and I'm gonna to compile it and make some tweaks if it's needed.

osdnk commented:
Oh, and I would be grateful if you manage to make a switch variable to turn on/off nots :)

zemiret commented:
Okay. The structure has changed a bit, since I've added checking whether notifications are enabled or disabled, etc.

So answering the questions:
1. Handling offline notifications - no problem. There is a function FCM.presentLocalNotification for which I've added two wrappers. One that does the same but checks if notifications are enabled before dispatching a notification, and a second being just a convenience one.
2. Setting SFI logo - no problem. I've added some custom icon (c_icon) to check if it works and it does. However there are some restrictions (and it's like this for all mobile applications). In ios notifications will use app icon always, unless you do some hacking to change it. In android, icons cannot have a background and since android 5.0 when the app is running, they will be displayed in color but when the app is in background they'll be all white (with background in some other color).
3. Switching notifications on and off. This is a bit problematic. What I've done disables notifications for every case, except one - someone has token and sends POST directly to that token. The notification will still be displayed (even if there was no code on the client side). And unfortunately this library doesn't support changing it purely on the client side, so we'd have to make a request to the server telling it not to send notifications anymore for this token. But for now I've omitted this part, since it might prove unneeded at all.
Ah, and I've added a little switch in settings to trigger notifications on / off.
